CMakeLists.txt

```diff
@@ -474,6 +476,32 @@ include_directories(api/include)
 
 add_subdirectory(api)
 
+if(WITH_OPENTRACING)
+  find_package(OpenTracing CONFIG QUIET)
+  if(NOT OpenTracing_FOUND)
+    set(OPENTRACING_DIR "third_party/opentracing-cpp")
+    message("Trying to use local ${OPENTRACING_DIR} from submodule")
+    if(EXISTS "${PROJECT_SOURCE_DIR}/${OPENTRACING_DIR}/.git")
+      set(SAVED_CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS})
+      set(CMAKE_CXX_FLAGS "")
```

Comment:
Why do we need set `CMAKE_CXX_FLAGS` to `""` here?
If we use `-DCMAKE_CXX_FLAGS=-D_GLIBCXX_USE_CXX11_ABI=0` , I think it's also need be passed into all targets.

Reply:
The main issue I encountered with using the passed down CXX flags is that the _third-party/opentracing-cpp_ build will fail due to `-Werror=redundant-move`, as observed in this run: https://github.com/open-telemetry/opentelemetry-cpp/actions/runs/3915878675/jobs/6694688717

I also considered disabling BUILD_MOCKTRACER which would effectively get rid of this compilation error and is arguably a much cleanerapproach, but I find it is more of a band aid than a fix, since any other flags that get added in the future might still interfere with the core opentracing-cpp build.

Reply:
I think it's better to disable this warning when include opentracing-cpp headers or add `-Wno-error=redundant-move` when include opentracing-cpp's headers.
Just reset `CMAKE_CXX_FLAGS` may break the ABI.

Reply:
Done. Thanks for the advise.